src/tabs/tab-header-group.component.ts

```diff
@@ -0,0 +1,287 @@
+import {
+	Component,
+	QueryList,
+	Input,
+	HostListener,
+	ContentChildren,
+	OnDestroy,
+	AfterContentInit,
+	ElementRef
+} from "@angular/core";
+
+import { TabHeader } from "./tab-header.component";
+import { Subscription } from "rxjs";
+
+@Component({
+	selector: "ibm-tab-header-group",
+	template: `
+	<nav
+		class="bx--tabs"
+		[ngClass]="{
+			'bx--skeleton': skeleton
+		}"
+		role="navigation"
+		[attr.aria-label]="ariaLabel"
+		[attr.aria-labelledby]="ariaLabelledby">
+		<div
+			class="bx--tabs-trigger"
+			tabindex="0"
+			(click)="showTabList()"
+			(keydown)="onDropdownKeydown($event)">
+			<a
+				href="javascript:void(0)"
+				class="bx--tabs-trigger-text"
+				tabindex="-1">
+				<ng-container *ngIf="!getSelectedTab().headingIsTemplate">
+					{{ getSelectedTab().heading }}
+				</ng-container>
+				<ng-template
+					*ngIf="getSelectedTab().headingIsTemplate"
+					[ngTemplateOutlet]="getSelectedTab().heading"
+					[ngTemplateOutletContext]="{$implicit: getSelectedTab().context}">
+				</ng-template>
+			</a>
+			<svg width="10" height="5" viewBox="0 0 10 5">
+				<path d="M0 0l5 4.998L10 0z" fill-rule="evenodd"></path>
+			</svg>
+		</div>
+		<ul
+			#tabList
+			[ngClass]="{
+				'bx--tabs__nav--hidden': !tabListVisible
+			}"
+			(keydown)="tabDropdownKeydown($event)"
+			class="bx--tabs__nav"
+			role="tablist">
+			<li role="presentation">
+				<ng-container *ngIf="contentBefore" [ngTemplateOutlet]="contentBefore"></ng-container>
+			</li>
+			<ng-content></ng-content>
+			<li role="presentation">
+				<ng-container *ngIf="contentAfter" [ngTemplateOutlet]="contentAfter"></ng-container>
+			</li>
+		</ul>
+	</nav>
+	 `
+})
+export class TabHeaderGroup implements AfterContentInit, OnDestroy {
+	/**
+	 * Set to 'true' to have tabs automatically activated and have their content displayed when they receive focus.
+	 */
+	@Input() followFocus: boolean;
+	/**
+	 * Set to `true` to put tabs in a loading state.
+	 */
+	@Input() skeleton = false;
+	/**
+	 * Sets the aria label on the nav element.
+	 */
+	@Input() ariaLabel: string;
+	/**
+	 * Sets the aria labelledby on the nav element.
+	 */
+	@Input() ariaLabelledby: string;
+
+	@Input() contentAfter;
+
+	@Input() contentBefore;
```

Comment:
no types for the 2 above?

Reply:
Do you think it should be of type TemplateRef<any>?